File: app/src/main/java/de/robv/android/xposed/XC_MethodHook.java
package de.robv.android.xposed;

import java.lang.reflect.Member;

/** Stub — replaced by LSPosed at runtime */
public class XC_MethodHook {
    public static class MethodHookParam {
        public Member method;
        public Object thisObject;
        public Object[] args;
        private Object result;
        private boolean returnEarly;
        public Object getResult() { return result; }
        public void setResult(Object result) { this.result = result; this.returnEarly = true; }
    }
    protected void beforeHookedMethod(MethodHookParam param) throws Throwable {}
    protected void afterHookedMethod(MethodHookParam param) throws Throwable {}
}

File: app/src/main/java/de/robv/android/xposed/XC_MethodHook.kt
package de.robv.android.xposed

import java.lang.reflect.Member

/** Stub — replaced by LSPosed at runtime */
open class XC_MethodHook {

    open class MethodHookParam {
        @JvmField var method: Member? = null
        @JvmField var thisObject: Any? = null
        @JvmField var args: Array<Any>? = null
        private var result: Any? = null
        private var returnEarly = false

        fun getResult(): Any? = result
        fun setResult(result: Any?) { this.result = result; returnEarly = true }
    }

    @Throws(Throwable::class)
    protected open fun beforeHookedMethod(param: MethodHookParam) {}

    @Throws(Throwable::class)
    protected open fun afterHookedMethod(param: MethodHookParam) {}
}
